From: Roman Smirnov <r.smirnov@omp.ru>
Subject: assoc_array: fix the return value in assoc_array_insert_mid_shortcut()
Date: Thu, 7 Mar 2024 10:17:17 +0300

Returning the edit variable is redundant because it is dereferenced right
before it is returned.  It would be better to return true.

Found by Linux Verification Center (linuxtesting.org) with Svace.

Link: https://lkml.kernel.org/r/20240307071717.5318-1-r.smirnov@omp.ru
Signed-off-by: Roman Smirnov <r.smirnov@omp.ru>
Reviewed-by: Sergey Shtylyov <s.shtylyov@omp.ru>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 lib/assoc_array.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/lib/assoc_array.c~assoc_array-fix-the-return-value-in-assoc_array_insert_mid_shortcut
+++ a/lib/assoc_array.c
@@ -938,7 +938,7 @@ static bool assoc_array_insert_mid_short
 		edit->leaf_p = &new_n0->slots[0];
 
 	pr_devel("<--%s() = ok [split shortcut]\n", __func__);
-	return edit;
+	return true;
 }
 
 /**
_
